telephone-line-segments.el

```diff
@@ -30,6 +30,23 @@
 (telephone-line-defsegment telephone-line-process-segment ()
   mode-line-process)
 
+(telephone-line-defsegment* telephone-line-vc-nobackend-segment ()
+  (if vc-mode
+      (substring-no-properties vc-mode (+ 1 (string-match "[-:@!?]" vc-mode)))
+    " - "))
+
+(telephone-line-defsegment* telephone-line-buffer-shortname-segment ()
+  ;; Avoids the padding in the regular "buffer only" segment
+  (buffer-name))
+
+(telephone-line-defsegment* telephone-line-position+region-segment ()
```

Comment:
I think it'd be better to make the region a separate segment instead. That way, it's easier to keep the position aligned no matter where you put the segments.

Reply:
The difference with the regular position segment is that this one doesn't like the position as percentage or "Bottom" etc.

Reply:
Well, why not? The percentage is fixed-width and wouldn't affect alignment -- and by the way alignment is why I think this should be a separate segment. If your position is right-aligned, you'll want region to be on the left so that it doesn't bounce around, but you'd want the opposite if you have it on the lhs.

Reply:
Spot on, didn't think of alignment.
I never liked the "bottom" or % thing in the mode line, never found it useful :)

Will submit the region-only segment to avoid the problem you mentioned!